sound/soc/sof/ipc4-loader.c

```diff
@@ -178,6 +282,30 @@ struct sof_ipc4_fw_module *sof_ipc4_find_module_by_uuid(struct snd_sof_dev *sdev
 		}
 	}
 
+	/*
+	 * Do not attempt to load external library in case the maximum number of
+	 * firmware libraries have been already loaded
+	 */
+	if ((lib_id + 1) == ipc4_data->max_libs_count) {
+		dev_err(sdev->dev,
+			"%s: Maximum allowed number of libraries reached (%u)\n",
+			__func__, ipc4_data->max_libs_count);
+		return NULL;
+	}
+
+	/* The module can not be found, try to load it as a library */
```

Comment:
we should move this comment before line 278

Reply:
Why? In the first loop we are checking the already loaded libraries. At this very point we start to look for the library from the filesystem as it was not found.